### ISSUE TITLE
Unhelpful error message when trying to use get-pip with Python 2.6

### ISSUE THREAD

**Issue by offbyone** [pypa/get-pip#24]
`get-pip.py` fails when run with Python 2.6. That's expected, but it should fail in a way that directs the user to `bootstrap.pypa.io/2.6/get-pip.py`

Sadly, due to cleared scrollback (and the fact that I was doing this with Ansible, which doesn't exactly give great error messages) I don't have the exact error message, but it *was* a SyntaxError on a set literal.

**Comment by hugovk** [comment]
Perhaps something like https://github.com/pypa/get-pip/pull/25?

**Comment by csm10495** [comment]
I get something like this on 2.6 with the 2.6 version.

```
c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\_vendor\urllib3\util\ssl_.py:339: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\_vendor\urllib3\util\ssl_.py:137: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '_ssl.c:507: error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version'),)': /packages/ac/95/a05b56bb975efa78d3557efa36acaf9cf5d2fd0ee0062060493687432e03/pip-9.0.3-py2.py3-none-any.whl
c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\_vendor\urllib3\util\ssl_.py:137: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '_ssl.c:507: error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version'),)': /packages/ac/95/a05b56bb975efa78d3557efa36acaf9cf5d2fd0ee0062060493687432e03/pip-9.0.3-py2.py3-none-any.whl
c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\_vendor\urllib3\util\ssl_.py:137: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '_ssl.c:507: error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version'),)': /packages/ac/95/a05b56bb975efa78d3557efa36acaf9cf5d2fd0ee0062060493687432e03/pip-9.0.3-py2.py3-none-any.whl
c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\_vendor\urllib3\util\ssl_.py:137: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '_ssl.c:507: error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version'),)': /packages/ac/95/a05b56bb975efa78d3557efa36acaf9cf5d2fd0ee0062060493687432e03/pip-9.0.3-py2.py3-none-any.whl
c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\_vendor\urllib3\util\ssl_.py:137: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '_ssl.c:507: error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version'),)': /packages/ac/95/a05b56bb975efa78d3557efa36acaf9cf5d2fd0ee0062060493687432e03/pip-9.0.3-py2.py3-none-any.whl
c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\_vendor\urllib3\util\ssl_.py:137: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
Exception:
Traceback (most recent call last):
  File "c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\basecommand.py", line 215, in main
    status = self.run(options, args)
  File "c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\commands\install.py", line 324, in run
    requirement_set.prepare_files(finder)
  File "c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\req\req_set.py", line 380, in prepare_files
    ignore_dependencies=self.ignore_dependencies))
  File "c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\req\req_set.py", line 620, in _prepare_file
    session=self.session, hashes=hashes)
  File "c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\download.py", line 821, in unpack_url
    hashes=hashes
  File "c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\download.py", line 659, in unpack_http_url
    hashes)
  File "c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\download.py", line 853, in _download_http_url
    stream=True,
  File "c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\_vendor\requests\sessions.py", line 521, in get
    return self.request('GET', url, **kwargs)
  File "c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\download.py", line 386, in request
    return super(PipSession, self).request(method, url, *args, **kwargs)
  File "c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\_vendor\requests\sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\_vendor\requests\sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\_vendor\cachecontrol\adapter.py", line 47, in send
    resp = super(CacheControlAdapter, self).send(request, **kw)
  File "c:\users\csm10495\appdata\local\temp\tmp2cs1sz\pip.zip\pip\_vendor\requests\adapters.py", line 506, in send
    raise SSLError(e, request=request)
SSLError: HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Max retries exceeded with url: /packages/ac/95/a05b56bb975efa78d3557efa36acaf9cf5d2fd0ee0062060493687432e03/pip-9.0.3-py2.py3-none-any.whl (Caused by SSLError(SSLError(1, '_ssl.c:507: error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version'),))
```

I don't think it works anymore on 2.6 since tls1.2 probably isn't supported there.

**Comment by brainwane** [comment]
Now that Python 2.6 is [no longer supported](https://www.python.org/dev/peps/pep-0361/), shall we close this issue?

**Comment by xavfernandez** [comment]
> Now that Python 2.6 is no longer supported, shall we close this issue?

Well, the `get-pip.py` for Python 2.6 lives in https://bootstrap.pypa.io/2.6/get-pip.py precisely precisely because Python 2.6 isn't supported: I think the point of this issue is "How can we make the legacy get-pip.py scripts more discoverable ?

#25 seems like a good idea but if I'm not mistaken, it would mean that the `get-pip.py` program will never be allowed to use python3 only syntax:

```
$ cat test.py 
import sys

if sys.version_info[0] == 2:
    print('Abort !')
    sys.exit(0)

async def toto():
  return 45

$ python2.7 test.py 
  File "test3.py", line 7
    async def toto():
            ^
SyntaxError: invalid syntax
```

It does not print `Abort!` (or any possibly useful message) in Python 2, it simply crashes.

Maybe a `https://bootstrap.pypa.io/README.txt` could (marginally) help ?

**Comment by pradyunsg** [comment]
This should be fixed with #100 now. At least in the newer templates.
